tests/integration/routes/test_errors.py

```diff
@@ -80,3 +80,287 @@ def test_errors_500_exception_during_error_handling(self):
 
                     self.assertStatusCode(500)
                     self.assertInBody("Sorry, there is a problem with this service")
+
+    def test_401_theme_default_cookie_exists(self):
+        # Given
+        self.launchSurvey("test_introduction")
+        self.assertInUrl("/questionnaire/introduction/")
+
+        # When
+        current_url = self.last_url
+        self.exit()
+        self.get(current_url)
+
+        # Then
+        self.assertStatusUnauthorised()
+        cookie = self.getCookie()
+        self.assertEqual(cookie.get("theme"), "default")
+        self.assertInBody(
+            '<p>You will need to <a href="http://upstream.url/sign-in/logout">sign back in</a> to access your account</p>'
+        )
+
+    def test_401_theme_social_cookie_exists(self):
+        # Given
+        self.launchSurvey("test_theme_social")
+        self.assertInUrl("/questionnaire/radio/")
+
+        # When
+        current_url = self.last_url
+        self.saveAndSignOut()
+        self.get(current_url)
+
+        # Then
+        self.assertStatusUnauthorised()
+        cookie = self.getCookie()
+        self.assertEqual(cookie.get("theme"), "social")
+        self.assertInBody(
+            '<p>To access this page you need to <a href="https://rh.ons.gov.uk/sign-in/logout">re-enter your access code</a>.</p>'
+        )
+
+    def test_401_no_cookie(self):
+        # Given
+        self.launchSurvey("test_introduction")
+        self.assertInUrl("/questionnaire/introduction/")
+
+        # When
+        current_url = self.last_url
+        self.exit()
+        self.deleteCookieAndGetUrl(current_url)
+
+        # Then
+        self.assertStatusUnauthorised()
+        self.assertInBody(
+            [
+                (
+                    '<p>If you are completing a business survey, you need to sign back in to <a href="https://surveys.ons.gov.uk/sign-in/logout">your account</'
+                    "a>.</p>"
+                ),
+                '<p>If you started your survey using an access code, you need to <a href="https://rh.ons.gov.uk/sign-in/logout">re-enter your code</a>.</p>',
```

Comment:
Worth referencing these links using the constants instead of  hardcoding, we will so have to change the RH URL so minimise the no. of changes. Same for the other tests also.

Reply:
Added, cheers!